### ISSUE TITLE
Enum POJO's are not being generated correctly with javaType

### ISSUE THREAD

**Issue by ctcoulter** [joelittlejohn/jsonschema2pojo#794]
I'm having an issue with enums being generated as classes instead of enums. Take the simple example provided below. 

If I run the following command, the enum is being generated as a class instead of an enum.
```
jsonschema2pojo -s bookshelf.schema.json -t tempout -a none -R
```

If i remove the javaType off the Genre schema, it will generate correctly as an enum. The issue is that if I don't provide the javaType, it will generate with an incorrect name (GenreSchema) or as a nested class inside the Book POJO.

Any help or insight would be greatly appreciated. Thanks!

**Bookshelf**
```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "type": "object",
    "id": "urn:jsonschema:Bookshelf",
    "javaType": "Bookshelf",
    "properties": {
        "books": {
            "type": "array",
            "items": {
                "$ref": "book.schema.json"
            }
        }
    }
}
```
**Book**
```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "type": "object",
    "id": "urn:jsonschema:Book",
    "javaType": "Book",
    "properties": {
        "name": {
            "type": "string"
        },
        "genre": {
            "$ref": "genre.schema.json"
        }
    }
}
```
**Genre**
```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "id": "urn:jsonschema:Genre",
    "type": "string",
    "javaType": "Genre",
    "enum": [
        "FANTASY",
        "SCIFI",
        "HISTORY"
    ]
}
```

**Comment by joelittlejohn** [comment]
I think you are experiencing this:

https://github.com/joelittlejohn/jsonschema2pojo/issues/194

Can you try again but include a package name in `"javaType": "Genre"`?

**Comment by man-csl** [comment]
@joelittlejohn  
I also have the same problem but unfortunately changing schema files is not an option in my case as I am getting them from an official specification: https://www.openchargealliance.org/protocols/ocpp-201/

Appreciate if you could let me know a workaround at-least.

**Comment by omidziaee** [comment]
I have the same issue as @joelittlejohn  I am working on the same schemas and when I build the project it creates all the enums as classes and it pops up a compile error. Any help appriciate.

**Comment by joelittlejohn** [comment]
This is caused by two problems:

1. When no package name is provided in javaType, we should be adding a package name. This was not working, but will be fixed in the next version (1.1.0) because #1103 has been fixed.
2. There is a bug in CodeModel and when enums are generated into the default/root package they get created as classes instead of enums. 

So you can work around this by adding a package name to the `javaType` value, or wait until 1.1.0 is release (which will likely be very soon).
